### PR TITLE
Add casks for 4 Unity modules

### PR DESCRIPTION
**Pull request by JesseTG**
- Unity for macOS (IL2CPP)
- Unity for AppleTV
- Unity for Linux (Mono and IL2CPP)

~~I haven't gone through most of the list yet, but I'm working on it now. It's been a while since I used my Mac so I need to upgrade everything first. I just wanted to submit this PR before I forgot.~~

Okay, I'm done now (with one exception, see below).

---

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free. **All passed, including the pre-existing modules after making the recommended changes.**
- [x] `brew style --fix {{cask_file}}` reports no offenses. **All pass (including pre-existing modules) except for one. See remarks [below](#issuecomment-739334598).**

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md). **I followed the same naming pattern as the other Unity modules.**
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask). **The other Unity modules are in this repo.**
- [x] `brew audit --new-cask {{cask_file}}` worked successfully. ~~**Partially done. Passes for all except base Unity, AppleTV, and Android casks.**~~ **All done, but the dependency on an external appcast means that the Actions workflow may need to be re-run to reflect this.**
- [x] `brew install --cask {{cask_file}}` worked successfully. **All passed, including pre-existing modules.**
- [x] `brew uninstall --cask {{cask_file}}` worked successfully. **All passed, including pre-existing modules.**

### CODE REVIEW

File: Casks/unity-appletv-support-for-editor.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cask "unity-appletv-support-for-editor" do
+  version "2020.1.15f1,97d0ae02d19d"
+  sha256 "0a07791d5e6b4c0d0d41567a218c801550d414b47c088dfb8a88d77d8fc16b9d"
+
+  url "https://download.unity3d.com/download_unity/#{version.after_comma}/MacEditorTargetInstaller/UnitySetup-AppleTV-Support-for-Editor-#{version.before_comma}.pkg",
+      verified: "download.unity3d.com/download_unity/"
```
Comment (vitorgalvao) [inline Casks/unity-appletv-support-for-editor.rb:6]: This is the new approach. Please do it on the other casks as well.

**Review by vitorgalvao — DISMISSED** [review]

File: Casks/unity-appletv-support-for-editor.rb (hunk 1 above)
Comment (JesseTG) [reply Casks/unity-appletv-support-for-editor.rb:6]: Done. I also modified the original Unity modules accordingly.

**Comment by JesseTG** [comment]
There is one `brew style` offense that I think might be an edge case. Here's the output:

```
Jesse@jesse-mac Casks % brew style unity.rb unity-android-support-for-editor.rb unity-appletv-support-for-editor.rb unity-ios-support-for-editor.rb unity-linux-il2cpp-support-for-editor.rb unity-linux-support-for-editor.rb
unity-lumin-support-for-editor.rb unity-macos-il2cpp-support-for-editor.rb unity-webgl-support-for-editor.rb unity-windows-support-for-editor.rb
Inspecting 10 files
.......C..

Offenses:

unity-macos-il2cpp-support-for-editor.rb:9:9: C: Description shouldn't contain the platform.
  desc "macOS (IL2CPP) target support for Unity"
        ^^^^^

10 files inspected, 1 offense detected, 1 offense auto-correctable
Jesse@jesse-mac Casks %
```

Unity can build games for a variety of platforms, and has two runtimes; Mono and IL2CPP. The default for each desktop platform is Mono, and IL2CPP is installed separately. Not only that, but each desktop platform can produce Mono builds for the other two. I think this is worthy of an exception, what about you?

**Comment by JesseTG** [comment]
Whoops, looks like Unity released a new version under my feet. I'll update the modules accordingly.

**Comment by JesseTG** [comment]
Uh, wait, it seems like some (but not all) of the modules don't have a corresponding entry in the appcast file (`releases-darwin.json`). What should I do?

**Comment by miccal** [comment]
Thank you for your great contribution to `homebrew-cask` @JesseTG, well done.

**Review by miccal — APPROVED** [review]

**Comment by JesseTG** [comment]
My pleasure, @miccal.

**Pull request merged**